contracts/Pool.sol

```diff
@@ -272,6 +342,22 @@ abstract contract Pool {
         ReceiverWeight[] calldata updatedReceivers,
         ReceiverWeight[] calldata updatedProxies
     ) public suspendPayments {
+        setReceiversInternal(updatedReceivers, updatedProxies);
+    }
+
+    /// @notice Sets the weight of the provided receivers and proxies of the sender of the message.
+    /// The weight regulates the share of the amount sent on every block
+    /// that each of the sender's receivers and proxies get.
+    /// Setting a non-zero weight for a new receiver or
+    /// a new proxy adds it to the list of the sender's receivers.
+    /// Setting zero as the weight for a receiver or a proxy
+    /// removes it from the list of the sender's receivers.
+    /// @param updatedReceivers The list of the updated receivers and their new weights
+    /// @param updatedProxies The list of the updated proxies and their new weights
+    function setReceiversInternal(
+        ReceiverWeight[] calldata updatedReceivers,
+        ReceiverWeight[] calldata updatedProxies
+    ) public {
```

Comment:
Should this really be public?

Reply:
Great catch! Absolutely not, it's a critical bug :scream: I've reviewed all the public functions and found `setAmountPerBlockInternal` with the same issue.